src/buildstream/downloadablefilesource.py

```diff
@@ -145,7 +221,22 @@ class DownloadableFileSource(Source):
     def configure(self, node):
         self.original_url = node.get_str("url")
         self.ref = node.get_str("ref", None)
-        self.url = self.translate_url(self.original_url)
+
+        extra_data = {}
+        self.url = self.translate_url(self.original_url, extra_data=extra_data)
+        self.auth_header_format = extra_data.get("auth-header-format")
```

Comment:
Fail if `extra_deps` contains unknown keys?

Reply:
I'm generally in agreement here, the `extra_data` is much like `public-data` and provides some _"extra context"_ which Source plugins _"might observe"_.

Otherwise, as @abderrahim points out, we will need SourceMirror instances to be very precisely targeting sources very specifically.